go/oasis-node/cmd/common/metrics/disk.go

```diff
@@ -0,0 +1,107 @@
+// Package metrics implements a prometheus metrics service.
+package metrics
+
+import (
+	"fmt"
+	"os"
+	"path/filepath"
+	"sync"
+
+	"github.com/prometheus/client_golang/prometheus"
+	"github.com/prometheus/procfs"
+	"github.com/spf13/viper"
+
+	"github.com/oasislabs/oasis-core/go/oasis-node/cmd/common"
+)
+
+const (
+	MetricDiskUsageBytes   = "oasis_worker_disk_usage_bytes"
+	MetricDiskReadBytes    = "oasis_worker_disk_read_bytes"
+	MetricDiskWrittenBytes = "oasis_worker_disk_written_bytes"
+)
+
+var (
+	diskUsageGauge          prometheus.Gauge
+	diskIOReadBytesGauge    prometheus.Gauge
+	diskIOWrittenBytesGauge prometheus.Gauge
+
+	diskServiceOnce sync.Once
+)
+
+type diskService struct {
+	ResourceMicroService
```

Comment:
The embedded `ResourceMicroService` interface is never set when `diskService` is constructed in `NewDiskService`? 

I don't think you even have a use case for embedding the interface here, so just dropping this should be fine. Similar for some of the other metric services.